internal/controller/pod_controller_test.go

```diff
@@ -320,8 +318,52 @@ var _ = Describe("CatGate controller", func() {
 					numSchedulable += 1
 				}
 			}
-			// 1 pod is already scheduled, but it is not running, so 1 pod should be scheduled
-			g.Expect(numSchedulable).To(Equal(1))
+			// 1 pod is already scheduled, but it is not running,
+			// so 2(capacity: 2, numImagePullingPods: 2) pod should be scheduled
+			g.Expect(numSchedulable).To(Equal(2))
+		}).Should(Succeed())
+	})
+
+	It("Should more Pods be scheduled if images are present in a node, not limited to the number of Pods.", func() {
+		testName := "already-images-in-node"
+		namespace := &corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: testName,
+			},
+		}
+		err := k8sClient.Create(ctx, namespace)
+		Expect(err).NotTo(HaveOccurred())
+
+		for i := 0; i < 10; i++ {
```

Comment:
Is the number of nodes correct?
Shouldn't we check that the no limit is applied in the following situations?

- Num of Pods > Num of Nodes
- All nodes have image cache.

Reply:
I fixed it with abc73c6f955dd0403ce9583dec5d2ade374b122e.